crates/mysten-metrics/src/lib.rs

```diff
@@ -81,3 +97,27 @@ macro_rules! spawn_monitored_task {
         tokio::task::spawn(mysten_metrics::monitored_future!(tasks, $fut))
     };
 }
+
+#[macro_export]
+macro_rules! monitored_scope {
+    ($name: expr) => {{
```

Comment:
If we have a 'name', perhaps we don't even need location at this point?

Locations change as code evolve, so they are not going to be used in dashboards or anything.

Even for monitored_futures, etc, I would prefer to just go with explicit name instead of deriving location from file/line

And cherry on top is that if we don't need location then this entire thing can turn into regular function

Reply:
Good point. I removed the location label for monitored_scope, and converted the macro into a function.